critiquebrainz/db/users.py

```diff
@@ -70,3 +73,543 @@ def get_many_by_mb_username(usernames):
+    return user
+
+
+def get_by_mbid(musicbrainz_id):
+    """Get user by musicbrainz_id
+
+    Args:
+        musicbrainz_id(str): Musicbrainz ID of the User
+
+    Returns:
+        Dictionary with the following structure:
+        {
+            "id": (uuid)
+            "display_name": (str)
+            "email": (str)
+            "created": (datetime)
+            "musicbrainz_id": (str)
+            "show_gravatar": (bool)
+            "is_blocked": (bool)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id, display_name, email, created,
+                   musicbrainz_id, show_gravatar, is_blocked
+            FROM "user"
+            WHERE musicbrainz_id = :musicbrainz_id
+        """), {
+            "musicbrainz_id": musicbrainz_id
+        })
+        row = result.fetchone()
+    return dict(row) if row else None
+
+
+def get_or_create(display_name, musicbrainz_id, **kwargs):
+    """Get user from display_name and musicbrainz_id
+
+    Args:
+        display_name(str): display_name of the user
+        musicbrainz_id(str): Musicbrainz ID of the user
+        email(str, optional): email of the user,
+        show_gravatar(bool, optional): whether to show gravatar(default: false)
+        is_blocked(bool, optional): whether user is blocked(default: false)
+
+    Returns:
+        Dictionary with the following structure:
+        {
+            "id": (uuid)
+            "display_name": (str)
+            "email": (str)
+            "created": (datetime)
+            "musicbrainz_id": (str)
+            "show_gravatar": (bool)
+            "is_blocked": (bool)
+        }
+    """
+    user = get_by_mbid(musicbrainz_id)
+    if not user:
+        user = create(display_name, musicbrainz_id=musicbrainz_id, **kwargs)
+    return user
+
+
+def get_count():
+    """Returns the total number of users
+
+    Returns:
+        count: (int)
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT count(*)
+              FROM "user"
+        """))
+        count = result.fetchone()[0]
+    return count
+
+
+def list(limit=None, offset=0):
+    """Returns the list of users of critiquebrainz
+
+    Args:
+        limit(int, optional): Number of users to be returned
+        offset(int, optional): Number of initial users to skip
+
+    Returns:
+        List of dictionaries of users with the following structure:
+        {
+            "id": (uuid)
+            "display_name": (str)
+            "email": (str)
+            "created": (datetime)
+            "musicbrainz_id": (str)
+            "show_gravatar": (bool)
+            "is_blocked": (bool)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id, display_name, email, created,
+                   musicbrainz_id, show_gravatar, is_blocked
```

Comment:
Each row name should be on a new line. https://github.com/metabrainz/guidelines/blob/master/Python.md#sql